invenio_banners/administration/banners.py

```diff
@@ -0,0 +1,229 @@
+                "Other - all the rest types of massages in a gray color."
+            ),
+            "options": [
+                {"title_l10n": "Info", "id": "info"},
+                {"title_l10n": "Warning", "id": "warning"},
+                {"title_l10n": "Other", "id": "other"},
+            ],
+        },
+        "active": {
+            "order": 6,
+            "text": _("Active"),
+            "description": _("Set to true to display a banner, false - to disable."),
+        },
+        "created": {"order": 7},
+        "updated": {"order": 8},
+    }
+
+
+class BannerCreateView(AdminResourceCreateView):
+    """Configuration for Banner create view."""
+
+    name = "banner_create"
+    url = "/banners/create"
+    resource_config = "banners_resource"
+    pid_path = "id"
+    api_endpoint = "/banners"
+    title = "Create Banner"
+
+    list_view_name = "banners"
+
+    form_fields = {
+        "start_datetime": {
+            "order": 1,
+            "text": _("Start DateTime"),
+            "description": _(
+                "DateTime input format: yyyy-mm-dd hh:mm:ss."
+                "Set to current or future date/time to delay the banner."
+            ),
+            "placeholder": _("YYYY-MM-DD hh:mm:ss"),
+        },
+        "end_datetime": {
+            "order": 2,
+            "text": _("End DateTime"),
+            "description": _(
+                "DateTime input format: yyyy-mm-dd hh:mm:ss. Date/time "
+                "to make the banner inactive. Empty value will make "
+                "the banner active until manually disabled via the active flag."
+            ),
+            "placeholder": _("YYYY-MM-DD hh:mm:ss"),
+        },
+        "message": {
+            "order": 3,
+            "text": _("Message"),
+            "description": _(
+                "Message to be displayed on the banner. HTML format is supported."
+            ),
+            "rows": 10,
+        },
+        "url_path": {
+            "order": 4,
+            "text": _("URL path"),
+            "description": _(
+                "Enter the URL path (including the first /) to define in "
+                "which part of the site the message will be active. For "
+                "example, if you enter `/records`, any URL starting with "
+                "`/records` will return an active banner (/records, "
+                "/records/1234, etc...). Empty value will make the banner "
+                "active for any URL."
+            ),
+        },
+        "category": {
+            "order": 5,
+            "text": _("Category"),
+            "description": _(
+                "Banner category. Info option displays the banner "
+                "with information message in a blue color. "
+                "Warning - a warning information in an orange color. "
+                "Other - all the rest types of messages in a gray color."
+            ),
+            "options": [
+                {"title_l10n": "Info", "id": "info"},
```

Comment:
Is there a way we can avoid hard-coding these banner types? E.g. retrieve them from a [config](https://github.dev/inveniosoftware/invenio-banners/blob/a2610fca3bf4de25b4927374ad89a8576c72a673/invenio_banners/config.py#L12)
I am not totally sure if we can do something like in [communities](https://github.com/inveniosoftware/invenio-communities/blob/72c72d8341f48f5db723674b3ab20fb95b6ace7a/invenio_communities/administration/communities.py#L64).

Reply:
Hey! I asked this question to myself as well and it seems like it's not possible (or at least not that easy), because this config is outside of `application context` and can not reach the config we need.
The method in the example of the communities that you sent is not used anywhere, so is probably invalid.